json.go

```diff
@@ -47,6 +48,120 @@ type NearLightClientBlockView struct {
 	Id      string
 }
 
+type BlockHeaderLite struct {
+	InnerLite     NearInnerLightView `json:"inner_lite"`
+	InnerRestHash string             `json:"inner_rest_hash"`
+	PrevBlockHash string             `json:"prev_block_hash"`
+}
+
+type MetaData struct {
+	GasProfile *string `json:"gas_profile"`
+	Version    uint64  `json:"version"`
+}
+type Status struct {
+	SuccessReceiptId string `json:"SuccessReceiptId"`
+}
+type Outcome struct {
+	ExecutorId  string   `json:"executor_id"`
+	GasBurnt    uint64   `json:"gas_burnt"`
+	Logs        []string `json:"logs"`
+	MetaData    MetaData `json:"metadata"`
+	ReceiptIds  []string `json:"receipt_ids"`
+	Status      Status   `json:"status"`
+	TokensBurnt string   `json:"tokens_burnt"`
+}
+type Proof []struct {
+	Direction string `json:"direction"`
+	Hash      string `json:"hash"`
+}
+type OutcomeProof struct {
+	BlockHash string  `json:"block_hash"`
+	Id        string  `json:"id"`
+	Outcome   Outcome `json:"outcome"`
+	Proof     Proof   `json:"proof"`
+}
+
+type ResultX struct {
+	BlockHeaderLite  BlockHeaderLite `json:"block_header_lite"`
+	BlockProof       Proof           `json:"block_proof"`
+	OutcomeProof     OutcomeProof    `json:"outcome_proof"`
+	OutcomeRootProof Proof           `json:"outcome_root_proof"`
+}
+type RpcResponse struct {
+	Id      string  `json:"id"`
+	Jsonrpc string  `json:"jsonrpc"`
+	Result  ResultX `json:"result"`
+}
+
+func GetProof(bp Proof) []nearprimitive.MerklePathItem {
+
+	merklePath := []nearprimitive.MerklePathItem{}
+	for i := 0; i < len(bp); i++ {
+		path_item_hash := &nearprimitive.CryptoHash{}
+		err := path_item_hash.TryFromRaw(base58.Decode(bp[i].Hash))
+		if err != nil {
+			fmt.Printf("Failed to decode hash to base58: %s", err)
+		}
+		if bp[i].Direction == "Right" {
+			merklePath = append(merklePath, nearprimitive.MerklePathItem{Hash: nearprimitive.MerkleHash(*path_item_hash), Direction: nearprimitive.Right})
+		} else {
+			merklePath = append(merklePath, nearprimitive.MerklePathItem{Hash: nearprimitive.MerkleHash(*path_item_hash), Direction: nearprimitive.Left})
+		}
+	}
+	return merklePath
+}
+
+func GetOutcomeProof(response string) (nearprimitive.OutcomeProof, []nearprimitive.MerklePathItem) {
+	bp := RpcResponse{}
+	err := json.Unmarshal([]byte(response), &bp)
+	if err != nil {
+		fmt.Printf("Failed to unmarshal RpcResponse: %s", err)
+	}
+	receipt_ids := []nearprimitive.CryptoHash{}
+	single_receipt := nearprimitive.CryptoHash{}
+	for i := 0; i < len(bp.Result.OutcomeProof.Outcome.ReceiptIds); i++ {
+		err = single_receipt.TryFromRaw(base58.Decode(bp.Result.OutcomeProof.Outcome.ReceiptIds[i]))
+		receipt_ids = append(receipt_ids, single_receipt)
+		// err = receipt_ids[i].TryFromRaw(base58.Decode(bp.Result.OutcomeProof.Outcome.ReceiptIds[i]))
+		if err != nil {
+			fmt.Printf("Failed to get receipt Id: %s", err)
+		}
+	}
+
+	token_burnt, _, _ := num.U128FromString(bp.Result.OutcomeProof.Outcome.TokensBurnt)
+
+	//know how to convert this
```

Comment:
This comment does not make sense?

Reply:
will remove this! was just for my understanding